docs/modules/ROOT/pages/erc20.adoc

```diff
@@ -1,420 +1,102 @@
-The contract should first import the ERC20 methods and the extended logic from the https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.6.1/src/openzeppelin/security/pausable/library.cairo[Pausable library] i.e. `Pausable.pause`, `Pausable.unpause`.
-Next, the contract should expose the methods with the extended logic therein like this:
-
-[,cairo]
-----
-@external
-func transfer{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-    recipient: felt, amount: Uint256
-) -> (success: felt) {
-    Pausable.assert_not_paused();                 // imported extended logic
-    return ERC20.transfer(recipient, amount);     // imported library method
-}
-----
-
-Note that extensibility does not have to be only library-based like in the above example.
-For instance, an ERC20 contract with a pausing mechanism can define the pausing methods directly in the contract or even import the `pausable` methods from the library and tailor them further.
-
-Some other ways to extend ERC20 contracts may include:
-
-* Implementing a minting mechanism.
-* Creating a timelock.
-* Adding roles such as owner or minter.
-
-For full examples of the extensibility pattern being used in ERC20 contracts, see <<presets,Presets>>.
-
-== Presets
-
-The following contract presets are ready to deploy and can be used as-is for quick prototyping and testing.
-Each preset mints an initial supply which is especially necessary for presets that do not expose a `mint` method.
-
-=== ERC20 (basic)
-
-The https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.6.1/src/openzeppelin/token/erc20/presets/ERC20.cairo[`ERC20`] preset offers a quick and easy setup for deploying a basic ERC20 token.
-
-=== ERC20Mintable
-
-The https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.6.1/src/openzeppelin/token/erc20/presets/ERC20Mintable.cairo[`ERC20Mintable`] preset allows the contract owner to mint new tokens.
+:components: https://community.starknet.io/t/cairo-1-contract-syntax-is-evolving/94794#extensibility-and-components-11[Components]
 
-=== ERC20Pausable
+WARNING: This is currently unsafe, because storage members could clash among used contracts if not reviewed carefully. Extensibility will be revisited after {components} are introduced.
 
-The https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.6.1/src/openzeppelin/token/erc20/presets/ERC20Pausable.cairo[`ERC20Pausable`] preset allows the contract owner to pause/unpause all state-modifying methods i.e.
-`transfer`, `approve`, etc.
-This preset proves useful for scenarios such as preventing trades until the end of an evaluation period and having an emergency switch for freezing all token transfers in the event of a large bug.
+ERC20 contracts can be extended by utilizing the contract's `_unsafe_new_contract_state()`.
+The unsafe contract state allows access to the `ERC20Impl`.
+This enables `IERC20` methods to include additional behaviors.
```

Comment:
I think is better to have the warning below the paragraph, because I think is not clear what "This" is referring to when we haven't presented the unsafe mechanism.

I would also change the wording from "This is currently unsafe, because storage members..." to "This mechanism is unsafe because storage members..."